app/lib/actions/katello/host/upload_profiles.rb

```diff
@@ -35,11 +35,32 @@ def rescue_strategy
 
         def import_module_streams(payload, host)
           enabled_payload = payload.map do |profile|
-            profile.slice("name", "stream", "version", "context", "arch") if profile["status"] == "enabled"
+            profile.slice("name", "stream", "version", "context", "arch").with_indifferent_access if profile["status"] == "enabled"
           end
           enabled_payload.compact!
+          # Get the library instances
+          bound_library_instances = host.content_facet.bound_repositories.map(&:library_instance_or_self)
 
-          ::Katello::Pulp::Consumer.new(host.content_facet.uuid).upload_module_stream_profile(enabled_payload)
+          query_name_streams = enabled_payload.map do |profile|
+            ::Katello::ModuleStream.where(profile.slice(:name, :stream))
+          end
```

Comment:
can we put the query_name_streams.inject(&:org) here?  i think it could easily be missed down below.  I originally thought you had forgot to put it in

Reply:
updated